jsapp/js/components/processing/sidebar/sidebarDisplaySettings.tsx

```diff
@@ -22,6 +33,25 @@ export default function SidebarDisplaySettings() {
     store.getDisplays(activeTab)
   );
 
+  function getInitialFields() {
+    if (!props.assetContent?.survey) {
+      return [];
+    }
+
+    const questionsList = getFlatQuestionsList(props.assetContent.survey, 0)
+      .filter((question) => !(question.name === store.currentQuestionName))
+      .map((question) => {
+        // We make an object to show the question label to the user but use the
+        // name internally so it works with duplicate question labels
+        return {name: question.name, label: question.label};
+      });
+
+    return questionsList;
+  }
+
+  const [selectedFields, setSelectedFields] = useState(getInitialFields());
+  const [hiddenFields, setHiddenFields] = useState<string[]>();
```

Comment:
> I have also noticed that I don't see a code that is reading the hidden fields from the store when opening display settings… (to be continued in different comment (above))

Should be added correctly now

Reply:
> You don't need to store hidden fields at all. Simply move `const hiddenList = …` to inside `applyFieldsSelection` function.
> 
> It would be something like this:
> 
> ```ts
>   function applyFieldsSelection() {
>     const hiddenFields: string[] = getCheckboxes()
>       .filter((checkbox) => !checkbox.checked)
>       .map((checkbox) => checkbox.name) || [];
>     store.setHiddenSidebarQuestions(hiddenFields);
>   }
> ```
> 
> And `resetFieldsSelection` function would become:
> 
> ```ts
>   function resetFieldsSelection() {
>     setSelectedFields(getInitialFields());
>     applyFieldsSelection();
>   }
> ```

This fix mostly worked, I left a comment in the code about the second part and why I had to set it to `[]` explicitly instead. There may be an even better solution but I think that is fine too, lmk